e2e/container/run_test.go

```diff
@@ -34,6 +35,22 @@ func TestRunAttachedFromRemoteImageAndRemove(t *testing.T) {
 	golden.Assert(t, result.Stderr(), "run-attached-from-remote-and-remove.golden")
 }
 
+// Regression test for https://github.com/docker/cli/issues/5053
+func TestRunInvalidEntrypointWithAutoremove(t *testing.T) {
+	environment.SkipIfDaemonNotLinux(t)
+
+	result := make(chan *icmd.Result)
+	go func() {
+		result <- icmd.RunCommand("docker", "run", "--entrypoint=invalidcommand", "--rm", fixtures.AlpineImage, "param")
+	}()
+	select {
+	case r := <-result:
+		r.Assert(t, icmd.Expected{ExitCode: 127})
+	case <-time.After(4 * time.Second):
+		t.Fatal("test took too long, shouldn't hang")
```

Comment:
Not sure if it makes the test cleaner (or the reverse), but `icmd.RunCommand` is a wrapper for `icmd.RunCmd` - the latter takes a `Command`, which can have a timeout on its own;

- https://github.com/gotestyourself/gotest.tools/blob/1fa4ef36538cd6021425ac34811bb7b20da3652a/icmd/command_test.go#L66-L70
- Or the `WithTimeOut` helper; https://github.com/gotestyourself/gotest.tools/blob/1fa4ef36538cd6021425ac34811bb7b20da3652a/icmd/ops.go#L12-L17

Really not sure if it's cleaner though; haven't tried yet what it looks like (just recalled that there was some way to set a timeout) 🙈

Reply:
I guess it would be something like;

```go
icmd.RunCmd(Command("docker", "run", "--entrypoint=invalidcommand", "--rm", fixtures.AlpineImage, "param"), WithTimeout(4 * time.Second))
```

Reply:
I generally prefer the way I had it (it's a pattern I'm used to in tests, and obvious where the timeout happens) but I guess there are some benefits? I don't really mind much, but I'll change it if it's preferred.

Reply:
I'm good with the current approach overall; I agree it's readable, so from that perspective 👍 

For transparency; I was curious if `icmd.RunCommand` had a variant that accepted a context; both with (future) OTEL tracing in tests in mind, and (possible) to use that for the timeout. 🥳 

But it looks like it currently doesn't 🤷‍♂️ but when I looked I (re)discovered that it has an option to set a timeout 😅 

> but I guess there are some benefits?

So .... maybe? A quick glance shows that it also handles killing runaway processes ... I guess? Not sure if it's super important, and haven't looked _really_ closely at the other bits it does;

https://github.com/docker/cli/blob/647ccf34334f76b7cc668f485143af4e11403c20/vendor/gotest.tools/v3/icmd/command.go#L270-L295